runtime/jcl/common/clsldr.cpp

```diff
@@ -30,36 +30,42 @@
 
 extern "C" {
 
-jboolean JNICALL Java_java_lang_ClassLoader_isVerboseImpl(JNIEnv *env, jclass clazz)
+jboolean JNICALL
+Java_java_lang_ClassLoader_isVerboseImpl(JNIEnv *env, jclass clazz)
 {
 	J9JavaVM *javaVM = ((J9VMThread *) env)->javaVM;
 
 	return ( (javaVM->verboseLevel & VERBOSE_CLASS) == VERBOSE_CLASS );
 }
 
-
-jclass JNICALL 
+jclass JNICALL
 Java_java_lang_ClassLoader_defineClassImpl(JNIEnv *env, jobject receiver, jstring className, jbyteArray classRep, jint offset, jint length, jobject protectionDomain)
 {
+	J9VMThread *currentThread = (J9VMThread *)env;
+	jclass result = defineClassCommon(env, receiver, className, classRep, offset, length, protectionDomain, 0, NULL);
```

Comment:
Why did you refactor this code to define the class before doing the verification of the name?  Once the class has been defined, it gets added to classloader tables and the like - I would expect other threads to be able to find it as though it was successfully defined even though it might fail below now.

Reply:
Whether the class is exempt from the usual name rules is dependent on whether it extends `MagicAccessorImpl`: that information isn't available before the call to `
defineClassCommon`. I think you're right, though, if the name is bad, it ought to be removed from those tables.

Reply:
Those class cannot be found because `Class.forName()` also makes use of `verifyQualifiedName()`.

Reply:
Need to check the resolveClassRef path as well

Reply:
I've restored validation of the name before the call, but deferred throwing the exception until we know the class is not exempt.